content/browser/storage_partition_impl.cc

```diff
@@ -528,6 +533,17 @@ StoragePartitionImpl* StoragePartitionImpl::Create(
       geofencing_manager.get(), host_zoom_level_context.get(),
       navigator_connect_context.get(),
       platform_notification_context.get());
+#else
+  StoragePartitionImpl* storage_partition = new StoragePartitionImpl(
+      context, partition_path, quota_manager.get(), appcache_service.get(),
+      filesystem_context.get(), database_tracker.get(),
+      dom_storage_context.get(), indexed_db_context.get(),
+      cache_storage_context.get(), service_worker_context.get(),
+      webrtc_identity_store.get(), special_storage_policy.get(),
+      geofencing_manager.get(), host_zoom_level_context.get(),
+      navigator_connect_context.get(),
+      nullptr);
+#endif
```

Comment:
minimal the DISABLE_NOTIFICATIONS block:

   StoragePartitionImpl* storage_partition = new StoragePartitionImpl(
       context, partition_path, quota_manager.get(), appcache_service.get(),
       filesystem_context.get(), database_tracker.get(),
       dom_storage_context.get(), indexed_db_context.get(),
       cache_storage_context.get(), service_worker_context.get(),
       webrtc_identity_store.get(), special_storage_policy.get(),
       geofencing_manager.get(), host_zoom_level_context.get(),
       navigator_connect_context.get(),
#ifndef DISABLE_NOTIFICATIONS
       platform_notification_context.get());
#else
       nullptr);
#endif